code/modules/mob/living/carbon/damage_procs.dm

```diff
@@ -0,0 +1,191 @@
+
+
+/mob/living/carbon/apply_damage(damage, damagetype = BRUTE, def_zone = null, blocked = 0)
+	var/hit_percent = (100-blocked)/100
+	if(!damage || hit_percent <= 0)
+		return 0
+
+	var/obj/item/bodypart/BP = null
+	if(islimb(def_zone)) //we specified a bodypart object
+		BP = def_zone
+	else
+		if(!def_zone)
+			def_zone = ran_zone(def_zone)
+		BP = get_bodypart(check_zone(def_zone))
+		if(!BP)
+			BP = bodyparts[1]
```

Comment:
One of the most important line: When we specify nothing or a def_zone (instead of an actual bodypart object) and we can't find the corresponding limb we default to the chest (because it can never be removed) instead of just stopping the proc and dealing no damage.

This fixes, among other things, not doing any damage when shooting someone while target a limb that the mob doesn't have (but still seeing a 'hit' message).